Jenkinsfile

```diff
@@ -89,7 +89,66 @@ try {
     parallel(tasks)
 
     common.setResultIfNotSet(Constants.JSUCCESS)
-} 
+
+
+    node('runPylint')  {
+        stage('runPylint')
+        {
+            try {
+              step('runPylint') {
+                sh """
+                   pylint --exit-zero --output-format=parseable --reports=n tenable > reports/pylint_tenable.log
+                   pylint --exit-zero --output-format=parseable --reports=n tests > reports/pylint_tests.log
+                   cat reports/pylint_tenable.log
+                   cat reports/pylint_tests.log
+                """
+            } catch(ex){
+                throw ex
+            } finally {
+                if (fileExists ('reports/pylint_tenable.log')) {
+                       //result needs to outpyt, will test
+                       result =  recordIssues(
+                            enabledForFailure: true,
+                            tool: pyLint(pattern: '**/pylint.log'),
+                            unstableTotalAll: 20,
+                            failedTotalAll: 30,)
+                                                                }
+                if (fileExists ('reports/pylint_test.log')) {
+                       //result needs to outpyt, will test
+                       result =  recordIssues(
+                            enabledForFailure: true,
+                            tool: pyLint(pattern: 'reports/pylint_*.log'),
+                            unstableTotalAll: 20,
+                            failedTotalAll: 30,)
+                                                              }
+                    }
+        }
+    }
+
+    node('runPyPi')  {
+        stage('runPyPi')
+        {
+            try {
+              step('runPyPi') {
+                  String prodOrTest = env.BRANCH_NAME == 'master' ?  'prod' : 'test'
+                  withCredentials([[$class : 'UsernamePasswordMultiBinding',
```

Comment:
I struggle a bit with the upload logic. Every possible branch will upload the results to the test repo. If the version number is the same on the potential branches, you will have a yoyo effect if two branches are under active development at the same time.

Reply:
We will upload only one time when version change occurs. Then file exist, it will not upload.
ie why --skip-existing

Reply:
So that means, the first build on a new version that runs on a branch on CI uploads the build results .... if you find an issue with this version, you need to make a version change to get a new test version uploaded? 

That sounds odd to me.

Reply:
Only the first time, we use. We expect no bugs in the existing version when we release that version.
ie right. If there is any issues , we fix and release new version. ie what we are following currently

Reply:
In the old logic you only pushed on build on master, no you push on each branch build. That's very different - but it's your product.

Reply:
We dont publish everytime Ralph, We push only in New version with Master, and not for tests.
ie why we dont have test branch. I had agreed to put as it is . So this code works only for Master branch build with new version. ie our requirement

Reply:
I have to disagree. As the code is written every build is pushing. The only difference between master and every other branch is that it is using different credentials.

Reply:
If you only want to push on master, there should be an if (env.BRANCH_NAME == 'master') .... around the whole stage.

Reply:
We will not need for all branches. This is for just the PyPI logic for the Master branch, and before we release, we do test in a Test branch. 

So for both , I asked Lev to give credentials, With that it needs to run. We never push for every branch. Which is not our requirement

Reply:
What you decribe above is not implemented in the logic.

Reply:
ie what it implements because of the logic of --skip-existing. ie why Ralph. that keyword takes place, if that version is present, dont upload. So automatically only one time. If that is test credentials or production

Reply:
@rboergertenable   I have followed your suggestions.  

- Removed steps as it was not required 
- Made Pylint as task, will run as part of parallel tasks
- pylint_* file patterns removed for now. If all works fine in testing, I will make it pylint_*  with one  call to recordIssues

Reply:
@rboergertenable   I have followed your advice and feedback.  Can you please review see if you can approve,